modules/exploits/linux/upnp/belkin_wemo_upnp_exec.rb

```diff
@@ -82,17 +83,35 @@ def check
     )
 
     if res && res.code == 200 && res.body.include?('urn:Belkin:device:')
-      vprint_good('Wemo-enabled device detected')
-      return CheckCode::Appears
+      vprint_status('Wemo-enabled device detected')
+    else
+      vprint_error('This does not appear to be a Wemo-enabled device')
+      return CheckCode::Safe
     end
 
-    CheckCode::Safe
+    version_text = res.get_xml_document.at('firmwareVersion').text
```

Comment:
This needs a `nil` check. I'll fix it.

```
[1] pry(#<Msf::Modules::Exploit__Linux__Upnp__Belkin_wemo_upnp_exec::MetasploitModule>)> res = Rex::Proto::Http::Response.new
=> #<Rex::Proto::Http::Response:0x00007f90a965e578
 @auto_cl=true,
 @body="",
 @bufq="",
 @chunk_max_size=10,
 @chunk_min_size=1,
 @code=200,
 @count_100=0,
 @headers={},
 @inside_chunk=false,
 @message="OK",
 @proto="1.1",
 @state=1,
 @transfer_chunked=false>
[2] pry(#<Msf::Modules::Exploit__Linux__Upnp__Belkin_wemo_upnp_exec::MetasploitModule>)> res.get_xml_document.at('firmwareVersion').text
NoMethodError: undefined method `text' for nil:NilClass
from (pry):6:in `__pry__'
[3] pry(#<Msf::Modules::Exploit__Linux__Upnp__Belkin_wemo_upnp_exec::MetasploitModule>)>
```

Reply:
For reference:

https://github.com/rapid7/metasploit-framework/blob/7394b5bf2507c17cb255ee658074fb2a75c740b2/modules/auxiliary/admin/wemo/crockpot.rb#L91-L93